format-parquet/src/main/java/io/cdap/plugin/format/parquet/input/ParquetInputFormatProvider.java

```diff
@@ -80,7 +81,13 @@ public Schema detectSchema(FormatContext context, InputFiles inputFiles) throws
       }
       try (ParquetFileReader parquetFileReader = ParquetFileReader.open(new HadoopInputFile(inputFile), readOptions)) {
         MessageType parquetSchema = parquetFileReader.getFooter().getFileMetaData().getSchema();
-        return new AvroToStructuredTransformer().convertSchema(new AvroSchemaConverter().convert(parquetSchema));
+        Schema schema = new AvroToStructuredTransformer().convertSchema(
+                new AvroSchemaConverter().convert(parquetSchema));
+        List<Schema.Field> fieldsList = schema.getFields();
+        if (conf.getPathField() != null) {
+          fieldsList.add(Schema.Field.of(conf.getPathField(), Schema.of(Schema.Type.STRING)));
+        }
+        return Schema.recordOf("Parquet", fieldsList);
```

Comment:
use `schema.getRecordName()` as the schema name.

To remove duplicate code, it would be better to add methods to the parent class 

```
Schema addPathField(Schema dataSchema) {
  List<Schema.Field> fields = new ArrayList<>(dataSchema.getFields());
  addPathField(fields);
  return Schema.recordOf(dataSchema.getRecordName(), fields);
}

void addPathField(List<Schema.Field> dataFields) {
  String pathField = conf.getPathField();
  if (pathField != null) {
    dataFields.add(Schema.Field.of(conf.getPathField(), Schema.of(Schema.Type.STRING)));
  }
}
```

and refactor all of these classes as well as the existing TextInputFormatProvider and BlobInputFormatProvider to use these common methods.